es6/snapshot.js

```
@@ -16,7 +16,13 @@ function snapshotObject(obj, origRef, context) {
     if (ref.hasOwnProperty(key)) foundRefKeys++;
 
     const referenceStateForKey = ref[key];
-    const clonedStateForKey = snapshotHelper(obj[key], referenceStateForKey, context);
+    let clonedStateForKey;
+    // In some cases,
+    // properties listed in `Object.keys` may actually throw Errors on read
+    try {
+      clonedStateForKey = snapshotHelper(obj[key], referenceStateForKey, context);
+    // eslint-disable-next-line no-empty
+    } catch (e) { }
```

To make sure we limit the error swallowing to just the property access, I'd prefer this:
```js
    let value;
    try {
      value = obj[key];
    } catch (e) {
      // Ignore errors generated accessing properties returned by `Object.keys()`.
    }
    const clonedStateForKey = snapshotHelper(value, referenceStateForKey, context);
```

---

Good stuff, thanks!